files/tk_metrics

```diff
@@ -146,42 +172,42 @@ end
 
 filename = Time.now.utc.strftime('%Y%m%dT%H%M%SZ') + '.json'
 
-HOSTS.each do |host|
+config[:hosts].each do |host|
   begin
     timestamp = Time.now
     dataset = {'timestamp' => timestamp.utc.iso8601, 'servers' => {}}
     hostkey = host.gsub('.', '-')
 
-    status_output   = get_status_endpoint(host, PORT, SSL)
-    dataset['servers'][hostkey] = {METRICS_TYPE => status_output}
+    status_output   = get_status_endpoint(host, config[:"metrics-port"], config[:ssl])
+    dataset['servers'][hostkey] = {config[:"metrics-type"] => status_output}
 
-    unless METRICS.empty? then
-      metrics_array = retrieve_additional_metrics(host, PORT, METRICS, PE_VERSION, SSL)
+    unless config[:"additional-metrics"].empty? then
+      metrics_array = retrieve_additional_metrics(host, config[:"metrics-port"], config[:"additional-metrics"], config[:"pe-version"], config[:ssl])
 
       metrics_array.each do |metric_hash|
         metric_name = metric_hash['name']
         metric_data = metric_hash['data']
 
-        dataset['servers'][hostkey][METRICS_TYPE][metric_name] = metric_data
+        dataset['servers'][hostkey][config[:"metrics-type"]][metric_name] = metric_data
       end
     end
 
-    dataset['servers'][hostkey][METRICS_TYPE]['error'] = $error_array
-    dataset['servers'][hostkey][METRICS_TYPE]['error_count'] = $error_array.count
-    dataset['servers'][hostkey][METRICS_TYPE]['api-query-start'] = timestamp.utc.iso8601
-    dataset['servers'][hostkey][METRICS_TYPE]['api-query-duration'] = Time.now - timestamp
+    dataset['servers'][hostkey][config[:"metrics-type"]]['error'] = $error_array
+    dataset['servers'][hostkey][config[:"metrics-type"]]['error_count'] = $error_array.count
+    dataset['servers'][hostkey][config[:"metrics-type"]]['api-query-start'] = timestamp.utc.iso8601
+    dataset['servers'][hostkey][config[:"metrics-type"]]['api-query-duration'] = Time.now - timestamp
 
     json_dataset = JSON.pretty_generate(dataset)
 
-    unless OUTPUT_DIR.nil? then
-      Dir.chdir(OUTPUT_DIR) do
+    unless config[:"output-dir"] == false then
```

Comment:
Yep. With a default value of False. Part of the option definition on line 35. Could probably change it to a symbol like `:undef`, should have roughly the same effect. This if-statement is just checking to see if the option has a valid value, or if it's the not-nil-but-not-valid default.